sockeye/data_io.py

```diff
@@ -396,79 +396,47 @@ def statistics(self):
                               average_len_target_per_bucket=self.mean_len_target_per_bucket,
                               length_ratio_stats_per_bucket=self.length_ratio_stats_per_bucket)
 
-
-def shard_data(source_fnames: List[str],
-               target_fnames: List[str],
-               source_vocabs: List[vocab.Vocab],
-               target_vocabs: List[vocab.Vocab],
-               num_shards: int,
-               buckets: List[Tuple[int, int]],
-               length_ratio_mean: float,
-               length_ratio_std: float,
-               output_prefix: str):
+def create_shards(source_fnames: List[str],
+                  target_fnames: List[str],
+                  num_shards: int,
+                  output_prefix: str):
     """
-    Assign int-coded source/target sentence pairs to shards at random.
+    Assign source/target sentence pairs to shards at random.
 
     :param source_fnames: The path to the source text (and optional token-parallel factor files).
     :param target_fnames: The path to the target text (and optional token-parallel factor files).
-    :param source_vocabs: Source vocabulary (and optional source factor vocabularies).
-    :param target_vocabs: Target vocabulary (and optional target factor vocabularies).
     :param num_shards: The total number of shards.
-    :param buckets: Bucket list.
-    :param length_ratio_mean: Mean length ratio.
-    :param length_ratio_std: Standard deviation of length ratios.
     :param output_prefix: The prefix under which the shard files will be created.
-    :return: Tuple of source (and source factor) file names, target (and target factor) file names
-             and statistics for each shard, as well as global statistics.
+    :return: Tuple of source (and source factor) file names and target (and target factor) file names for each shard.
     """
     os.makedirs(output_prefix, exist_ok=True)
     sources_shard_fnames = [[os.path.join(output_prefix, C.SHARD_SOURCE % i) + ".%d" % f for i in range(num_shards)]
                             for f in range(len(source_fnames))]
     targets_shard_fnames = [[os.path.join(output_prefix, C.SHARD_TARGET % i) + ".%d" % f for i in range(num_shards)]
                             for f in range(len(target_fnames))]
 
-    data_stats_accumulator = DataStatisticsAccumulator(buckets, source_vocabs[0], target_vocabs[0],
-                                                       length_ratio_mean, length_ratio_std)
-    per_shard_stat_accumulators = [
-        DataStatisticsAccumulator(buckets, source_vocabs[0], target_vocabs[0], length_ratio_mean,
-                                  length_ratio_std) for shard_idx in range(num_shards)]
-
     with ExitStack() as exit_stack:
-        sources_shards = [[exit_stack.enter_context(smart_open(f, mode="wt")) for f in sources_shard_fnames[i]] for i in
+        sources_shards = [[exit_stack.enter_context(smart_open(f, mode="wb")) for f in sources_shard_fnames[i]] for i in
```

Comment:
When we read and write in binary mode we could potentially speed up the splitting. In the function here I see that we open the output files as binary, but do read/parse the files as unicode by using the sequence readers. Would this still speed things up compared to `wt`?

I would either:
1. Keep `wt` (unless there is indeed a speed advantage)
2. or try a version of split that reads/writes lines in binary without using the sequence reader to keep splitting minimal.

What do you think?

Reply:
There was a slight speedup with `wb` compared to `wt` in my experiments with the smaller dataset. I can run both versions on the larger dataset and if there is no significant speedup then I'll keep `wt`, otherwise I'll try (2).